### PR TITLE
Mandate node version for toolpad-app

### PR DESCRIPTION
**Pull request by Janpot**
For maintainers running Toolpad on their locally installed node

### CODE REVIEW

**Comment by Janpot** [comment]
🤔  right, netlify wants to install everything, even `toolpad-app` dependencies.

File: packages/toolpad-app/package.json [hunk 1]
```diff
@@ -137,5 +137,8 @@
   "typings": "./index.d.ts",
   "publishConfig": {
     "access": "public"
+  },
+  "engines": {
+    "node": ">=18"
```
Comment (oliviertassinari) [inline packages/toolpad-app/package.json:142]: Why so high?

```jsx
$ node --version
v16.16.0
```

**Comment by oliviertassinari** [comment]
> netlify wants to install everything, even toolpad-app dependencies.

https://github.com/mui/mui-toolpad/blob/3c603a5cc1abbfb9ef66572f184debcd45917246/netlify.toml#L11

File: packages/toolpad-app/package.json (hunk 1 above)
Comment (Janpot) [reply packages/toolpad-app/package.json:142]: > Why so high?

We could treat it as any other dependency in the project, and keep it up to date through renovatebot.

**Comment by oliviertassinari** [comment]
@Janpot what do you think about reopening to set Node.js 16 as the lower limit? This is what is recommended for most developers: https://nodejs.org/en/

**Comment by Janpot** [comment]
> what do you think about reopening to set Node.js 16 as the lower limit?

Not all versions in 16.x support all the features we use. (e.g. I think `AbortSignal.timeout` doesn't work pre 16.14). How do you feel about setting it to 16.18? Would there be a case for not keeping it up to date?

**Comment by oliviertassinari** [comment]
> How do you feel about setting it to 16.18? Would there be a case for not keeping it up to date?

It sounds great. It would fail upfront for developers, not having to wonder why AbortSignal.timeout is not defined.

**Comment by Janpot** [comment]
I'll set it at 16.17 for now. There's a circleci job runing on the docker machine executor and it only has 16.17.1 installed. We can look into fixing this part of CI in a separate effort.

**Pull request merged**